lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/stage/EmitDeletedByPrefix.java

```diff
@@ -0,0 +1,130 @@
+package com.kmwllc.lucille.pinecone.stage;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.pinecone.util.PineconeManager;
+import com.typesafe.config.Config;
+import dev.langchain4j.agent.tool.P;
+import io.pinecone.proto.ListItem;
+import io.pinecone.proto.ListResponse;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This stage retrieves all records from the Pinecone namespaces in Pinecone configs if the record's ID prefix matches the
+ * ID of the Lucille document passing through the stage AND if the document was already marked for deletion.
+ * These records are then emitted as Lucille Documents, which are also marked for deletion.
+ *
+ * Config Parameters:
+ * - dropOriginal (Boolean, Optional): if set to true, will mark documents in the pipeline that were marked for deletion
+ *   as dropped. Does not apply to emitted documents, only the original documents. Defaults to false
+ * - addPrefix (String, optional) : String that adds on to all document ids used for listing
+ *    e.g. if id of doc is "doc1" & addPrefix is "-v1", then this stage will retrieve and emit documents where their
+ *         id prefix is "doc1-v1".
+ */
+public class EmitDeletedByPrefix extends Stage {
+
+  private final PineconeManager pineconeManager;
+  private final Map<String, Object> namespaces;
+  private final boolean dropOriginal;
+  private final String addPrefix;
+  private final String deletionMarkerField;
+  private final String deletionMarkerFieldValue;
+
+  private static final Logger log = LoggerFactory.getLogger(EmitDeletedByPrefix.class);
+
+  public EmitDeletedByPrefix(Config config) {
+    super(config, new StageSpec()
+        .withOptionalProperties("dropOriginal", "addPrefix")
+        .withOptionalParents("namespaces")
+        .withRequiredProperties("apiKey", "deletionMarkerField", "deletionMarkerFieldValue", "index"));
+    this.pineconeManager = PineconeManager.getInstance(config.getString("apiKey"), config.getString("index"));
```

Comment:
How to "share" config values? It would be nice for stage to retrieve configs from other parts of the configs.

Reply:
config files can reference environment variables or system properties that are passed in.. so you the same config value can be used in multiple places...

Reply:
Along with referencing an env var or system property, you can define a variable in the conf file itself and then reference it in multiple places using substitutions. The syntax is explained here: https://github.com/lightbend/config?tab=readme-ov-file#uses-of-substitutions

So, if you set myVal=1 and you want to pass myVal into multiple stages, you would include myProp=${myVal} in the config for each stage